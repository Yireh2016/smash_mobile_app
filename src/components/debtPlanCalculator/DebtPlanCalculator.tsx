import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Slider from '@react-native-community/slider';

import theme from '../../theme/theme';
import { P1, H1, H3 } from '../texts/Texts';
import TableRow from '../../components/tableRow/TableRow';
import formatMoneyValue from '../../helpers/formatMoneyValue/formatMoneyValue';
import isEstimatedSavings from '../../helpers/isEstimatedSavings/isEstimatedSavings';

export interface DebtPlanCalculatorProps {
  isEditable: boolean;
  sliderControl: number;
  monthlyPayment: string | null;
  monthlyInput: string;
  onMonthlyInputChange: (arg0: number) => void;
  estimatedPayoffTime: string;
  estimatedSavings: string;
  estimatedPayoffDate: string | number;
}

const styles = StyleSheet.create({
  debtPlanLayout: {
    backgroundColor: theme.palette.primaryLight,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 14,
    paddingRight: 14,
    marginTop: 10,
    borderRadius: 10,
  },
  sliderContainer: {
    width: '100%',
  },
  sliderStyle: { width: '100%', height: 40 },
  monthlyPaymentLayout: {
    alignItems: 'center',
  },
  estimatedResultsTableContainer: {
    marginTop: 18,
    width: '100%',
  },
  tableRowContainer: {
    marginTop: 14,
  },
});

const DebtPlanCalculator: React.FunctionComponent<DebtPlanCalculatorProps> = ({
  isEditable,
  sliderControl,
  monthlyPayment,
  monthlyInput,
  onMonthlyInputChange,
  estimatedPayoffTime,
  estimatedSavings,
  estimatedPayoffDate,
}) => {
  return (
    <View style={styles.debtPlanLayout}>
      <View style={styles.monthlyPaymentLayout}>
        <View>
          <P1>Monthly payment</P1>
        </View>
        <View testID="monthlyPaymentContainer">
          <H1 color={theme.palette.secondary}>
            {isEditable
              ? formatMoneyValue(parseFloat(monthlyInput))
              : monthlyPayment || ''}
          </H1>
        </View>
        {isEditable && (
          <View style={styles.sliderContainer}>
            <Slider
              testID="Slider"
              value={sliderControl}
              onSlidingComplete={(value) => onMonthlyInputChange(value)}
              style={styles.sliderStyle}
              minimumValue={0}
              maximumValue={10}
              minimumTrackTintColor={theme.palette.secondary}
              maximumTrackTintColor={theme.palette.secondary}
            />
          </View>
        )}
        <View style={styles.estimatedResultsTableContainer}>
          <View style={styles.tableRowContainer}>
            <TableRow
              columns={[
                {
                  component: <P1>Est. pay off time</P1>,
                  align: 'left',
                },
                {
                  component: <H3>{estimatedPayoffTime}</H3>,
                  align: 'right',
                },
              ]}
            />
          </View>
          {isEstimatedSavings(estimatedSavings) && (
            <View style={styles.tableRowContainer}>
              <TableRow
                columns={[
                  {
                    component: <P1>Est. savings</P1>,
                    align: 'left',
                  },
                  {
                    component: <H3>{estimatedSavings}</H3>,
                    align: 'right',
                  },
                ]}
              />
            </View>
          )}
          <View style={styles.tableRowContainer}>
            <TableRow
              columns={[
                {
                  component: <P1>Est. pay off date</P1>,
                  align: 'left',
                },
                {
                  component: <H3>{estimatedPayoffDate}</H3>,
                  align: 'right',
                },
              ]}
            />
          </View>
        </View>
      </View>
    </View>
  );
};

export default DebtPlanCalculator;
