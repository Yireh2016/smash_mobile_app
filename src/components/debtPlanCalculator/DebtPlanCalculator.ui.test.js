import * as React from 'react';
import renderer from 'react-test-renderer';
import DebtPlanCalculator from './DebtPlanCalculator';
describe('', () => {
  it('DebtPlanCalculator w estimated Savings renders correctly', () => {
    const props = {
      isEditable: true,
      sliderControl: 8,
      monthlyPayment: null,
      monthlyInput: '937',
      onMonthlyInputChange: () => jest.mock(),
      estimatedPayoffTime: '27 weeks',
      estimatedSavings: '$300 /yr',
      estimatedPayoffDate: 'Dec 12, 2020',
    };

    const Element = () => {
      return <DebtPlanCalculator {...props} />;
    };

    const elementRendered = renderer.create(<Element />).toJSON();
    expect(elementRendered).toMatchSnapshot();
  });

  it('DebtPlanCalculator with no estimated Savings renders correctly', () => {
    const props = {
      isEditable: true,
      sliderControl: 8,
      monthlyPayment: null,
      monthlyInput: '937',
      onMonthlyInputChange: () => jest.mock(),
      estimatedPayoffTime: '27 weeks',
      estimatedSavings: '$0 /yr',
      estimatedPayoffDate: 'Dec 12, 2020',
    };

    const Element = () => {
      return <DebtPlanCalculator {...props} />;
    };

    const elementRendered = renderer.create(<Element />).toJSON();
    expect(elementRendered).toMatchSnapshot();
  });
});
