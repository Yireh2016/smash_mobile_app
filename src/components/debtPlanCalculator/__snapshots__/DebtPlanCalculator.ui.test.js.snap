// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebtPlanCalculator w estimated Savings renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#142836",
      "borderRadius": 10,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
          }
        }
      >
        Monthly payment
      </Text>
    </View>
    <View
      testID="monthlyPaymentContainer"
    >
      <Text
        style={
          Object {
            "color": "#29b782",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 24,
          }
        }
      >
        $937
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <RNCSlider
        disabled={false}
        enabled={true}
        inverted={false}
        maximumTrackTintColor="#29b782"
        maximumValue={10}
        minimumTrackTintColor="#29b782"
        minimumValue={0}
        onChange={null}
        onRNCSliderSlidingComplete={[Function]}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={null}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        step={0}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "height": 40,
              "width": "100%",
            },
          ]
        }
        testID="Slider"
        thumbImage={null}
        value={8}
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 18,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Est. pay off time
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              27 weeks
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Est. savings
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              $300 /yr
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Est. pay off date
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              Dec 12, 2020
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`DebtPlanCalculator with no estimated Savings renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#142836",
      "borderRadius": 10,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
          }
        }
      >
        Monthly payment
      </Text>
    </View>
    <View
      testID="monthlyPaymentContainer"
    >
      <Text
        style={
          Object {
            "color": "#29b782",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 24,
          }
        }
      >
        $937
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <RNCSlider
        disabled={false}
        enabled={true}
        inverted={false}
        maximumTrackTintColor="#29b782"
        maximumValue={10}
        minimumTrackTintColor="#29b782"
        minimumValue={0}
        onChange={null}
        onRNCSliderSlidingComplete={[Function]}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={null}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        step={0}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "height": 40,
              "width": "100%",
            },
          ]
        }
        testID="Slider"
        thumbImage={null}
        value={8}
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 18,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Est. pay off time
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              27 weeks
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Est. pay off date
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              Dec 12, 2020
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
